pkgs/by-name/se/sequoia-wot/package.nix

```diff
@@ -0,0 +1,53 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitLab,
+  libclang,
+  pkg-config,
+  nettle,
+  openssl,
+  sqlite,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "sequoia-wot";
+  version = "0.11.0";
+
+  src = fetchFromGitLab {
+    owner = "sequoia-pgp";
+    repo = "sequoia-wot";
+    rev = "v${version}";
+    hash = "sha256-qSf2uESsMGUEvAiRefpwxHKyizbq5Sst3SpjKaMIWTQ=";
+  };
+
+  cargoPatches = [
+    ./01-cargo-lock.patch
+    ./02-cargo-toml.patch
+  ];
+
+  cargoHash = "sha256-ZPVvRulCXG81eTuL2oettUrrFWEmZk8VzD9YB+P+wVc=";
+
+  nativeBuildInputs = [
+    pkg-config
+    rustPlatform.bindgenHook
+  ];
+
+  buildInputs = [
+    libclang
+    nettle
+    openssl
+    sqlite
+  ];
+
+  buildFeatures = [ "sequoia-openpgp/default" ];
+
+  # Panic caused by accessing GNUPGHOME in https://gitlab.com/sequoia-pgp/sequoia-wot/-/blob/v0.11.0/tests/gpg.rs?ref_type=tags#L25
```

Comment:
Did you try setting `GNUPGHOME` in a `preCheck` hook?

Reply:
I set

```nix
  preCheck = ''
    export GNUPGHOME=$(mktemp -d)
    export RUST_BACKTRACE=1
  '';
```

and failed with

```log
sequoia-wot> ---- gpg_trust_roots stdout ----
sequoia-wot> thread 'gpg_trust_roots' panicked at tests/gpg.rs:25:30:
sequoia-wot> called `Result::unwrap()` on an `Err` value: No such file or directory (os error 2)
sequoia-wot> Stack backtrace:
sequoia-wot>    0: anyhow::error::<impl core::convert::From<E> for anyhow::Error>::from
sequoia-wot>    1: gpg::gpg::export
sequoia-wot>    2: core::ops::function::FnOnce::call_once
sequoia-wot>    3: test::__rust_begin_short_backtrace
sequoia-wot>    4: test::types::RunnableTest::run
sequoia-wot>    5: test::run_test_in_process
sequoia-wot>    6: std::sys_common::backtrace::__rust_begin_short_backtrace
sequoia-wot>    7: core::ops::function::FnOnce::call_once{{vtable.shim}}
sequoia-wot>    8: std::sys::pal::unix::thread::Thread::new::thread_start
sequoia-wot>    9: start_thread
sequoia-wot>   10: __GI___clone3
sequoia-wot> stack backtrace:
sequoia-wot>    0: rust_begin_unwind
sequoia-wot>    1: core::panicking::panic_fmt
sequoia-wot>    2: core::result::unwrap_failed
sequoia-wot>    3: core::ops::function::FnOnce::call_once
sequoia-wot> note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
sequoia-wot> failures:
sequoia-wot>     gpg_trust_roots
sequoia-wot> test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```